tosfs/core.py

```diff
@@ -203,6 +203,55 @@ def info(
 
         return self._try_dir_info(bucket, key, path, fullpath)
 
+    def rmdir(self, path: str) -> None:
+        """Remove a directory if it is empty.
+
+        Parameters
+        ----------
+        path : str
+            The path of the directory to remove. The path should be in the format
+            `tos://bucket/path/to/directory`.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the directory does not exist.
+        NotADirectoryError
+            If the path is not a directory.
+        TosfsError
+            If the directory is not empty,
+             or the path is a bucket.
+
+        Examples
+        --------
+        >>> fs = TosFileSystem()
+        >>> fs.rmdir("tos://mybucket/mydir/")
+
+        """
+        path = self._strip_protocol(path).rstrip("/") + "/"
+        bucket, key, _ = self._split_path(path)
+        if not key:
+            raise TosfsError("Cannot remove a bucket using rmdir api.")
+
+        if not self.exists(path):
+            raise FileNotFoundError(f"Directory {path} not found.")
+
+        if not self.isdir(path):
+            raise NotADirectoryError(f"{path} is not a directory.")
+
+        if len(self._listdir(bucket, max_items=1, prefix=key.rstrip("/") + "/")) > 0:
```

Comment:
@openinx Put a solution that is better than the previous one, but it's uncertain whether there is a more efficient one.

Reply:
Essentially, the reason why I commented as not efficient is:   we call two self.infos , and at least one OBJECT LIST ( the object LIST the a limit 1000 QPS by default).  So for the `rmrdir`, it will easily reach the QPS limit. Also the latency of limit is much more higher than the normal HEAD (10ms vs 1ms ). 

But I don't want this block your further PRs, So I plan to merge this firstly.  We can improve it in the next following PRs. 

Thanks.